doc/release/technotes/README.libraries.rst

```diff
@@ -23,36 +26,111 @@ the ``chpl`` command line take precedence over any found through object
 library paths (``-L``).  Where there is a conflict, the last library
 specified takes precedence.
 
-When Chapel code is accessed as library code, it is the user's responsibility to
-ensure that modules are initialized before they are used.  For a module
-named ``<myModule>``, the corresponding initialization routine is
-named ``chpl__init_<myModule>``.  Therefore, for example, a C program that
-makes use of the standard Chapel module ``Random`` should
-call ``chpl__init_Random`` at some point before the Random interface is first
-used.
+When creating a library file from Chapel code, only those symbols with
+``export`` attached to them will be available from outside the library.  For
+example:
+
+.. code-block:: Chapel
+
+   // This function will be available to the library user
+   export proc foo() {
+     // Does something
+     ...
+   }
+
+   // but this function will not be
+   proc bar() {
+     // Does something else
+     ...
+   }
+
+When using a Chapel library file in C code, a fairly exact incantation is
+required.  First, a header file must be created for the library.  This does not
+happen through use of the Chapel ``--library`` flag during compilation!  If
+during compilation of the library the ``--savec <dir>`` flag is also used, the C
+definitions for the exported functions should be visible in a C file under
+*dir* of the same name as the library Chapel file.  A declaration version of
+these definitions should go into a ``.h`` file.  For instance, if one compiled
+a Chapel file ``foo.chpl`` like this:
+
+.. code-block:: sh
+
+   chpl --library --savec cDir --static -o libfoo foo.chpl
+
+
+then under ``cDir/foo.c`` one might see something resembling:
+
+.. code-block:: C
+
+   int64_t bar(void) {
+     ...
+   }
+
+   void baz(int64_t x) {
+     ...
+   }
+
+from which one would create a ``.h`` file:
 
 .. code-block:: C
 
-    // C code using the Chapel Random module.
-    int main()
-    {
-      _int64 seed = 0;
-      double rand;
+   int64_t bar(void);
+   void baz(int64_t x);
+
+It may be necessary to ``#include`` some Chapel headers, such as
+``chpl-string.h``, depending on the types used in your Chapel library.
+
+Next, if compiling dynamically, update the ``$LD_LIBRARY_PATH`` environment
+variable to include the directory where the new library file lives and the
+directory where the Chapel build lives.  The latter can be found by looking at
+the output of a ``$CHPL_HOME/util/printchplenv`` call and finding the
+appropriate directory under ``$CHPL_HOME/lib``; the directory name can be found
+by running ``$CHPL_HOME/util/printchplenv --runtime``.
+
+.. code-block:: sh
+
+   # Replace $PWD with the appropriate path to your library file if it isn't
+   # in the current directory
+   # Replace libDir with the directory we just found.
+   export LD_LIBRARY_PATH=$PWD:<libDir>:$LD_LIBRARY_PATH
```

Comment:
Now that we have the printchplenv call, could this be:
export LD_LIBRARY_PATH=$PWD:$CHPL_HOME/lib/\`$CHPL_HOME/util/printchplenv --runtime\`:$LD_LIBRARY_PATH